caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -46,6 +46,11 @@
 logger = alog.use_channel("TXT_EMB")
 error = error_handler.get(logger)
 
+# Avoid a warning by explicitly setting TOKENIZERS_PARALLELISM if not already set.
```

Comment:
More explanation...

Background detail:   This came up in a scenario where lazy loading is used.   So the model is always loaded on first request, and with this truncation code the tokenizer is always immediately used.  So this error message is consistent on first use.

```
2023-11-28T18:07:19.695712 [MODEL-SIZER :INFO:8166760128] <RUN62161564I> No configured model size multiplier found for model type 'standalone-model' for model 'example/model22'. Using default multiplier '10.000000'
2023-11-28T18:07:19.698317 [sentence_tra:INFO:11295420416] Load pretrained SentenceTransformer: /Users/markstur/caikit-nlp/models/example/model22/artifacts
2023-11-28T18:07:20.679813 [sentence_tra:INFO:11295420416] Use pytorch device: cpu
2023-11-28T18:07:20.679940 [MODEL-LOADER:INFO:11295420416] <RUN89713784I> Singleton cache: '{}'
huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...
To disable this warning, you can either:
        - Avoid using `tokenizers` before the fork if possible
        - Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)
```

So, the 2nd option is implemented here by setting the env if not already set.  The nice thing about this is that any deployment can set the environment if they don't want this default.

There is a simple alternative that might be less ugly(?).  If I just add a single call to model.encode("foo") at the end of load() that seems to avoid using tokenizers too early, because essentially the model (and tokenizer) is primed and ready.  This seems like the less straight-forward way to address this specific problem, but it is a nice single line of code and a quick model test to make sure the model is "ready".  Unfortunately with lazy load this warmup is on the first request anyway, but otherwise I might like some amount of warmup.